include/flox/resolver/environment.hh

```diff
@@ -59,8 +59,30 @@ FLOX_DEFINE_EXCEPTION( ResolutionFailureException,
 
 /* -------------------------------------------------------------------------- */
 
+/**
+ * @brief A pair of _install ID_ and locked flake URLs used to memoize failed
```

Comment:
suggestion blocking: I think this is just used to accumulate failures to print a descriptive error. I'm not quite sure what you mean by using it to memoize?